### PR TITLE
[MBL-864] Stripe Upgrade

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What

Part of maintenance to upgrade all libraries to latest versions. [2023 Technical Plan](https://kickstarter.atlassian.net/wiki/spaces/NT/pages/2398584833/2023+iOS+Technical+Engineering+Roadmap) lists Stripe because their have been many updates on Stripes side.

- They've added the address element, many fixes to the payment sheet (including more payment types), a customer sheet, etc.
- They've also enabled Apple Pay Later, which we should not have enabled in our app.

Entire release logs go from 22.8.4 -> 23.10.0 
https://github.com/stripe/stripe-ios/releases/tag/22.8.4

# 🤔 Why

Upkeep with iOS system improvements and features from Stripe.

# 🛠 How

- Started with 22.8.4.

- 23.0.0 was a breaking change because Stripe split their single framework into (Stripe, StripePaymentSheet, StripePaymentUI, StripeApplePay and a few others). [Details](https://github.com/stripe/stripe-ios/releases/tag/23.0.0).
 
  - Basically in app these means we just switch `import Stripe` to `import StripePaymentSheet` or leave it as is for non-payment sheet references. 
  - One change to fix our tests was to go from using `PaymentSheet.FlowController.PaymentOptionDisplayData` to `PaymentSheetPaymentOptionsDisplayData`. The former is a `Stripe` data structure that crashes in tests because finding a card brand causes a module reference to fail being found.
  - `PaymentSheet.FlowController.PaymentOptionDisplayData` is just an `UIImage` and `String` so we can make our own version and use it in-app easily.

- Updated Stripe in SPM to Up to Next Major (23.10.0 < 24.0.0).

- Also did a bit of clean up removing the `AddNewCardViewController` from `Settings` storyboard, a card brand type extension and an unused `PaymentCardTextField`

# ✅ Acceptance criteria

- [x] Ensure no changes to Apple Pay
- [x] Ensure no changes to Payment Sheet (iPad and iPhone) UI
- [x] Ensure that AirTable [script](https://airtable.com/app4015jjYVK28ISz/tbl37u86jwRoyl7tk/viwgBXyxtVBUO56hj?blocks=hide) that presents the Pledge Page Stripe Payment Sheet is run through and all tests pass.
- [x] Ensure that AirTable [script](https://airtable.com/app4015jjYVK28ISz/tblGUu51Oav245mSa/viwToXvsYzfykPlG9?blocks=hide) that presents the Stripe Payment Sheet in User Settings is run through and all tests pass.

### CODE REVIEW

**Review by scottkicks — APPROVED** [review]
All AC passed
- No changes to Apple Pay look to have changed
- No changes to Payment Sheet (iPad and iPhone) UI have changed
- Airtable scripts/tests pass

Just left a comment about the module imports you've updated. I think just a couple small changes could be made, but nothing major.

OOO the rest of the week after today, so use your best judgment to merge this if you make those changes and I'm already gone.

File: Library/ViewModels/PledgePaymentMethodsViewModelTests.swift
Comment (msadoon) [reply Library/ViewModels/PledgePaymentMethodsViewModelTests.swift:7]: Good point - tested removing `import Stripe` in favor of `import StripePaymentSheet` and tests still pass.
Corrections made.

Comment (msadoon) [reply Library/ViewModels/PledgePaymentMethodsViewModelTests.swift]: @scottkicks I wanted to call out this mistake in your last [pr](https://github.com/kickstarter/ios-oss/pull/1833) that caused a build crash. I only noticed it when trying to build the this branch to run on device.

It's easy to miss in the review because its' a change to the `.project.pbxproj` file, which is a massive directory basically so hard to track file membership of which file does in which framework. 

Turns out `GraphAPI.TriggerThirdPartyEventInput+TriggerThirdPartyEventInputTests.swift` and `TriggerThirdPartyEventInputTests.swift` were added to `KsApi` when they should be added to `KsApiTests`

Fixed [here](https://github.com/kickstarter/ios-oss/pull/1834/commits/6bf56866051654bb4ba3b451d2f86000b4de2902).

I had to go back to my fix graph schema pr, run it on device, see that it was working then work backwards to find that the pr above had an issue with file membership which caused the on-launch crash.

Just be careful with adding the files to the right frameworks because it effects the run-time scenario in the app, even if the code compiles. Not sure why tests still passed in CI though.

**Pull request merged**
